packages/components/addon/components/hds/table/index.js

```diff
@@ -179,4 +184,106 @@ export default class HdsTableIndexComponent extends Component {
       onSort(this.sortBy, this.sortOrder);
     }
   }
+
+  @action
+  onSelectionAllChange(headerCheckbox) {
+    const checkboxes = this.allTbodyCheckboxes();
+    this.setHeaderCheckboxAriaLabel(headerCheckbox);
+    checkboxes.forEach((checkbox) => {
+      checkbox.checked = headerCheckbox.checked;
+      this.setRowCheckboxAriaLabel(checkbox);
+    });
+
+    if (headerCheckbox.checked) {
+      this.selectedSelectionKeys = [...this.allSelectionKeys];
+    } else {
+      this.selectedSelectionKeys = [];
+    }
+
+    let { onSelectionChange } = this.args;
+    if (typeof onSelectionChange === 'function') {
+      onSelectionChange([...this.selectedSelectionKeys]);
+    }
+  }
+
+  @action
+  onSelectionRowChange(checkbox, selectionKey) {
+    if (checkbox.checked) {
+      this.selectedSelectionKeys.push(selectionKey);
+    } else {
+      this.selectedSelectionKeys = this.selectedSelectionKeys.filter(
+        (key) => key !== selectionKey
+      );
+    }
+    this.setRowCheckboxAriaLabel(checkbox);
+    this.setSelectAllState();
+    let { onSelectionChange } = this.args;
+    if (typeof onSelectionChange === 'function') {
+      onSelectionChange([...this.selectedSelectionKeys]);
+    }
+  }
+
+  @action
+  didInsertRowCheckbox(checkbox, selectionKey) {
+    this.allSelectionKeys.push(selectionKey);
+    if (checkbox.checked) {
+      this.selectedSelectionKeys.push(selectionKey);
+    }
+    this.setRowCheckboxAriaLabel(checkbox);
+    this.setSelectAllState();
+  }
+
+  @action
+  willDestroyRowCheckbox(selectionKey) {
+    this.allSelectionKeys = this.allSelectionKeys.filter(
+      (key) => key !== selectionKey
+    );
+    this.selectedSelectionKeys = this.selectedSelectionKeys.filter(
+      (key) => key !== selectionKey
+    );
+    this.setSelectAllState();
+  }
+
+  allTbodyCheckboxes() {
+    const tableBodyCheckboxes = document.querySelectorAll(
+      `#${this.tableId} tbody .hds-table__checkbox`
+    );
+    return Array.from(tableBodyCheckboxes);
+  }
+
+  setRowCheckboxAriaLabel(checkbox) {
+    if (checkbox.checked === true) {
+      checkbox.ariaLabel = this.args.deselectRowAriaLabel
+        ? `${this.args.deselectRowAriaLabel} ${checkbox.id}`
+        : `Deselect ${checkbox.id}`;
+    } else {
+      checkbox.ariaLabel = this.args.selectRowAriaLabel
+        ? `${this.args.selectRowAriaLabel} ${checkbox.id}`
+        : `Select ${checkbox.id}`;
+    }
+  }
```

Comment:
It's not meaningful enough to have the checkbox's generated ID as the aria-label value, so we're going to have to think about this one some more. 

Some potential options to consider:
- row count (select row #)
- row count out of total (select row # of ##) 
- use the row's TH (scope of row) (select row Nick Drake) 

Also ultimately, we want to allow them to set the aria-label themselves, which provides room to grow/provide translations as the value instead.

Reply:
Thanks, I did provide an option for translations with this though.

Reply:
Cristiano is making some related updates in his test branch so I'll wait to go over those with him before making further accessibility-related updates in this branch. Thanks for reviewing.

Reply:
Closing this thread, now the `aria-label` values are more meaningful